### PR TITLE
Allow Running PCT executable with Java 11

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Just a branch which integrates all bleeding edge patches. It should help us to quickly deliver patches for Java 11 testing without distracting the proper review cycles for the master branch.

CC @alecharp @kevinearls

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,7 +52,7 @@ demo-jdk8: plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar tmp/je
 demo-jdk11: plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar tmp/jenkins-war-$(JENKINS_VERSION).war print-java-home
 	# TODO Cleanup when/if the JAXB bundling issue is resolved.
 	# https://issues.jenkins-ci.org/browse/JENKINS-52186
-	java -jar plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar \
+	$(TEST_JDK_HOME)/bin/java -jar plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar \
```
Comment (oleg-nenashev) [inline Makefile]: This patch is for running PCT internal logic with Java 11

File: Makefile [hunk 2]
```diff
@@ -68,14 +70,18 @@ demo-jdk11-docker: tmp/jenkins-war-$(JENKINS_VERSION).war
...
 	     jenkins/pct
 
+# TODO: take other default directory to avoid collisions?
+# Runs the build with local source located in /work
 .PHONY: demo-jdk11-docker-src
 demo-jdk11-docker-src: tmp/jenkins-war-$(JENKINS_VERSION).war
 	docker run --rm -v maven-repo:/root/.m2 \
 	     -v $(shell pwd)/out:/pct/out \
-	     -v $(shell pwd)/work/$(PLUGIN_NAME):/pct/plugin-src:ro \
+	     -v $(LOCAL_SRC):/pct/plugin-src:ro \
```
Comment (oleg-nenashev) [inline Makefile]: Quality of life improvement (MOAR FEATURES)

File: Makefile [hunk 3]
```diff
@@ -50,7 +52,7 @@ demo-jdk8: plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar tmp/je
 demo-jdk11: plugins-compat-tester-cli/target/plugins-compat-tester-cli.jar tmp/jenkins-war-$(JENKINS_VERSION).war print-java-home
 	# TODO Cleanup when/if the JAXB bundling issue is resolved.
```
Comment (MRamonLeon) [inline Makefile]: Can we remove these comments?

File: README.md [hunk 4]
```diff
@@ -111,6 +111,16 @@ Currently Java 8 and Java 11 are bundled (JDK_VERSION= {8, 11}).
 
 Full list of options for JDK11 can be found [here](./Makefile).
 
+#### Running PCT with custom Java versions in Docker
+
+When using the docker image, it is possible to use `JDK_VERSION` variable to select 
+the version to use. The version needs to be bundled in the docker image. By specifying 
+`11`, the modules and jaxb libraries are also added to the command line.
```
Comment (MRamonLeon) [inline README.md]: The JAXB library is no more needed after the detached plugin. We can rephrase the last part of the paragraph.

File: Makefile (hunk 3 above)
Comment (alecharp) [reply Makefile]:
```suggestion
```

**Comment by jglick** [comment]
Presumably obsolete.

**Pull request closed without merging**